drivers/ipvlan/ipvlan_network.go

```diff
@@ -0,0 +1,163 @@
+package ipvlan
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/libnetwork/driverapi"
+	"github.com/docker/libnetwork/netlabel"
+	"github.com/docker/libnetwork/options"
+	"github.com/docker/libnetwork/types"
+)
+
+// CreateNetwork the network for the specified driver type
+func (d *driver) CreateNetwork(id string, option map[string]interface{}, ipV4Data, ipV6Data []driverapi.IPAMData) error {
+	// parse and validate the config and bind to networkConfiguration
+	config, err := parseNetworkOptions(id, option)
+	if err != nil {
+		return err
+	}
+	if id == "" {
+		return fmt.Errorf("Network ID cannot be empty")
+	}
+	config.ID = id
+	err = config.processIPAM(id, ipV4Data, ipV6Data)
+	if err != nil {
+		return err
+	}
+	// verify the macvlan mode from --ipvlan_mode option
+	switch config.IpvlanMode {
+	case modeL2:
```

Comment:
If you think it is worth to make the cange, FYI golang would allow:
```
case modelL2, "":
    // default to ipvlan L2 mode if --ipvlan_mode is empty
    config.IpvlanMode = modeL2
```